### DISCUSSION TITLE
EXC_BAD_ACCESS during deallocation

### DISCUSSION THREAD

**Discussion by paulstraw** [finnvoor/PlaydateKit#92] [Q&A] [answered]
I think this may actually be an upstream problem, but figured I'd raise here since there hasn't been any response to swiftlang/swift#74697 (and frankly I'm a little intimidated to comment there 😅). Perhaps someone Swiftier than me will have an answer.

Unlike the OP @tyetrask, I've also seen this happening with classes in my own code, not just structs. I haven't been able to get a minimal repro of that yet, though.

I've created a PlaydateKit-based repro from Tye's sample, not sure how helpful that will be: https://github.com/paulstraw/pdk-exc-bad-access

**Comment by finnvoor** [comment, marked as answer]
added some more info here: https://github.com/swiftlang/swift/issues/74697#issuecomment-2323304501

**Reply by tyetrask** [reply]
Thanks for that reply and non-PlaydateKit-based repro, @finnvoor 🙌🙌🙌! Super helpful!

**Comment by paulstraw** [comment]
I can confirm this is fixed for me as of the September 17, 2024 trunk development build. Thanks very much for your help pushing this along @finnvoor!
